src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -382,7 +382,13 @@ public static String randomFieldName(String part, int numStartChars, int numEndC
     {
         // use the characters that we know are encoded in fieldKeys plus characters that we know clients are using
         String chars = ALL_ILLEGAL_QUERY_KEY_CHARACTERS + " %()=+-[]_|*`'\":;<>?!@#^";
-        String randomFieldName = (randomString(numStartChars, exclusion, chars) + part + randomString(numEndChars, null, chars)).trim();
+
+        // Having double space is allowed in a field name but is a problem for automation.
+        // We render double spaces as a single space as column headers in grids and the like, but we maintain the double
+        // space in the name. This trips up helpers for various components. See Issue 52193 for details.
+        String randomFieldName = (randomString(numStartChars, exclusion, chars).replace("  ", " ") +
+                part +
+                randomString(numEndChars, null, chars).replace("  ", " ")).trim();
```

Comment:
This is really an issue with QueryGrid and EditableGrid (and others I'm sure). We should open an issue to track and prioritize a refactor of our tests and helpers to key off of field names (or fieldKey) instead of using field labels.